qemu/tests/qemu_guest_agent.py

```diff
@@ -980,6 +980,74 @@ def gagent_check_file_write(self, test, params, env):
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        error_context.context("Change guest-file related cmd to white list"
+                              " and get guest file name.")
+        session, tmp_file = self._guest_file_prepare()
+        content = "helloworld\n"
+
+        error_context.context("Create a new small file in guest", logging.info)
+        cmd_create_file = "echo helloworld > %s" % tmp_file
+        session.cmd(cmd_create_file)
+        error_context.context("Open guest file via guest-file-open with"
+                              " read only mode.", logging.info)
+        # default is read mode
+        ret_handle = int(self.gagent.guest_file_open(tmp_file))
+        error_context.context("Read the content and check the result via"
+                              " guest-file cmd", logging.info)
+        self._read_check(ret_handle, content)
+        self.gagent.guest_file_close(ret_handle)
+
+        error_context.context("Create a 200KB file in guest", logging.info)
+        process.run("dd if=/dev/urandom of=/tmp/big_file bs=1024 count=200")
+        self.vm.copy_files_to("/tmp/big_file", tmp_file)
+        error_context.context("Open the big guest file via guest-file-open with"
+                              " read only mode.", logging.info)
+        # default is read mode
+        ret_handle = int(self.gagent.guest_file_open(tmp_file))
+        error_context.context("Read the big file with big count number.", logging.info)
+        try:
+            self.gagent.guest_file_read(ret_handle, count=10000000000)
+        except guest_agent.VAgentCmdError as detail:
+            if not re.search('invalid for argument count', str(detail)):
+                self.test.fail("This is not the desired information: "
+                               "('%s')" % str(detail))
+        self.gagent.guest_file_seek(ret_handle, 0, 0)
+        try:
+            self.gagent.guest_file_read(ret_handle, count=1000000000)
+        except guest_agent.VAgentCmdError as detail:
+            if not re.search('Insufficient system resources exist to complete '
+                             'the requested service', str(detail)):
+                self.test.fail("This is not the desired information: "
+                               "('%s')" % str(detail))
+        self.gagent.guest_file_close(ret_handle)
+
+        error_context.context("Open a non existing file with read only mode.",
+                              logging.info)
+        try:
+            self.gagent.guest_file_open("none_exist_file")
+        except guest_agent.VAgentCmdError as detail:
+            res_linux = "No such file or directory"
+            res_windows = "system cannot find the file"
+            if res_windows not in str(detail) and res_linux not in str(detail):
+                self.test.fail("This is not the desired information: "
+                               "('%s')" % str(detail))
+
```

Comment:
It needs to delete the `tmp_file` in the end

Reply:
will modify,thanks.